lib/router/handler-info/unresolved-handler-info-by-object.js

```diff
@@ -25,7 +25,8 @@ var UnresolvedHandlerInfoByObject = subclass(HandlerInfo, {
   serialize: function(_model) {
     var model = _model || this.context,
         names = this.names,
-        handler = this.handler;
+        handler = this.handler,
+        serializer = this.serializer || (handler && handler.serialize);
```

Comment:
This part seems a bit odd. If the idea of moving the serializer to the route is to allow lazy-loading, then the handler will only be available if the asset containing it has been loaded. Unless `getSerializer` always returns a function with default behavior or we use a flag to toggle between the old and the new behavior, this could lead to a different serialization logic used when an asset is loaded and when it's not. Considering that moving away from Route#serialize is relatively trivial, I suggest that both modes aren't supported for a given instance of a Router.